### PR TITLE
NXP-30684: Shortened pah strategy by default for local blob store

### PR DESCRIPTION
**Pull request by yjulienne-nuxeo**

### CODE REVIEW

File: nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/blob/LocalBlobStoreConfiguration.java [hunk 1]
```diff
@@ -45,6 +45,15 @@
...
     public static final String DEFAULT_PATH = "binaries";
 
+    /** In the initialization properties, the property for the store path strategy type (full|shortened). */
+    public static final String PROP_PATH_STRATEGY_TYPE = "pathStrategyType";
+
+    public static final String PATH_STRATEGY_TYPE_FULL = "full";
+
+    public static final String PATH_STRATEGY_TYPE_SHORTENED = "shortened";
+
+    public static final String DEFAULT_PATH_STRATEGY_TYPE = PATH_STRATEGY_TYPE_SHORTENED;
```
Comment (kevinleturc) [inline nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/blob/LocalBlobStoreConfiguration.java]: I think we should keep PATH_STRATEGY_TYPE_FULL as default for 10.10.

Comment (efge) [reply nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/blob/LocalBlobStoreConfiguration.java]: Yes this is necessary for compat. But I even suggest not doing that part of the change at all, see my comment in https://github.com/nuxeo/nuxeo-lts/pull/331#discussion_r773973855

File: nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/blob/LocalBlobStoreConfiguration.java [hunk 2]
```diff
@@ -117,4 +126,29 @@ public BinaryManagerRootDescriptor getDescriptor(Path configFile, boolean create
...
+
+    public PathStrategy getPathStategy(KeyStrategy keyStrategy, Path dir) {
+        String pathStrategyType = getPathStrategyType();
+        if (keyStrategy.useDeDuplication()) {
+            switch (pathStrategyType) {
+            case PATH_STRATEGY_TYPE_FULL:
+                return new PathStrategySubDirs(dir, descriptor.depth);
+            case PATH_STRATEGY_TYPE_SHORTENED:
+            default:
+                return new PathStrategyShortenedSubDirs(dir, descriptor.depth);
```
Comment (kevinleturc) [inline nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/blob/LocalBlobStoreConfiguration.java]: The default switch case should be along with PATH_STRATEGY_TYPE_FULL.

File: nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/blob/PathStrategyShortened.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+import java.nio.file.Path;
+
+import org.apache.commons.codec.digest.DigestUtils;
+
+/**
+ * Extends the default {@link PathStrategy#safePath(String)} method to ensure resulting path and file name is not too
+ * long.
+ *
+ * @since 2021.13
```
Comment (ataillefer) [inline nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/blob/PathStrategyShortened.java]: 2021.14 or 10.10-HF57, same below

**Review by ataillefer — DISMISSED** [review]

**Review by kevinleturc — DISMISSED** [review]

**Review by kevinleturc — APPROVED** [review]

**Review by ataillefer — APPROVED** [review]

**Review by efge — APPROVED** [review]

**Pull request merged**